src/mpid/ch4/shm/xpmem/xpmem_init.c

```diff
@@ -92,10 +117,33 @@ int MPIDI_XPMEM_mpi_finalize_hook(void)
     MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIDI_XPMEM_FINALIZE_HOOK);
     MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDI_XPMEM_FINALIZE_HOOK);
 
+    if (MPIR_Process.comm_world->node_comm) {
```

Comment:
Can we add a comment here saying that the barrier ensures all local XPMEM-based data transfer is done?